### ISSUE TITLE
Memory leak after clicking Reset or Pre-def

### ISSUE THREAD

**Issue by Lurchicus** [Lurchicus/YALife#2]
I'm pretty sure I'm mismanaging the ILife and ISave arrays somewhere. Maybe even "Paper" (the custom bitmap). 

Dan Rhea

**Comment by Lurchicus** [comment]
I believe I've found most of the leaks except one, and I'm not sure it's a leak.

I ckick on "pre-def", the procedure called by the event handler opens an OpenFileDialog object (to select/open a .gol file). Before I even make a file selection I see a huge jump in memory allocation (about 30 MB), after selecting the file and closing the dialog, none of the 30 MB appears to be returned, the code that loads and parses the .gol file has little effect, even after selecting run. I suspect that some or all of the memory for OpenFileDialog is not being returned.

        /// <summary>
        /// Open a file dialog and locate a .gol file
        /// </summary>
        private void GetGolFile()
        {
            OpenFileDialog GetGolFile = new()
            {
                Filter = "GOL File|*.gol|All files|*.*",
                FilterIndex = 0,
                InitialDirectory = Application.ExecutablePath
            };
            if (GetGolFile.ShowDialog() == DialogResult.OK)
            {
                PredefinedFile = GetGolFile.FileName;
                UsingPredefined = true;
            }
            else
            {
                UsingPredefined = false;
            }
            GetGolFile.Dispose();
            GC.Collect();
        }

I expected the Dispose() of the OpenFileDialog to return at lease  some of the 30 MB.

**Comment by Lurchicus** [comment]
Well, I clicked the wrong button and destroyed a long comment on the issue (I also didn't mean to close it). 

I believe I have the leak fixed, but a call to OpenFileDialog grabs about 30 MB and doesn't return it when closed and disposed. Maybe a .NET 7 Win Forms bug? 

        /// <summary>
        /// Open a file dialog and locate a .gol file
        /// </summary>
        private void GetGolFile()
        {
            OpenFileDialog GetGolFile = new()
            {
                Filter = "GOL File|*.gol|All files|*.*",
                FilterIndex = 0,
                InitialDirectory = Application.ExecutablePath
            };
            if (GetGolFile.ShowDialog() == DialogResult.OK)
            {
                PredefinedFile = GetGolFile.FileName;
                UsingPredefined = true;
            }
            else
            {
                UsingPredefined = false;
            }
            GetGolFile.Dispose();
            GC.Collect();
        }

**Comment by Lurchicus** [comment]
So, it looks like I'm chasing a red herring. 

I switched over to the OpenFileDialog I had droped on LifeForm and got exactly the same memory usage. It looks like once
you have referenced the dialog, memory overhead goes up but doesn't get returned... so it's overhead not a leak? I'll check
it again when .NET 18 is released to be sure. 

The changes to to ILife and ISave as well as the bitmap "Paper" have cleared up the memory leaks.
